### ISSUE TITLE
Should "--sequential-training" be default for tesstrain?

### ISSUE THREAD

**Issue by wrznr** [tesseract-ocr/tesstrain#109]
Given the documentation at https://github.com/tesseract-ocr/tesseract/wiki/TrainingTesseract-4.00#randomized-training-data-and-sequential_training, it might be reasonable to use `--sequential-training` per default since we will never have multiple lines in a single image file.

**Comment by stweil** [comment]
Indeed, that seems to be reasonable. I think it won't change the outcome of the training process, but it should significantly reduce the memory use. The current training obviously caches all data from the lstmf files in main memory. Thus memory usage grows while the training proceeds until all training data was loaded. For a training set with about 300000 lines like the GT4HistOCR data the `lstmtraining` process uses about 5.4 GiB RAM. A host with only 4 MiB RAM would start swapping, resulting in slower training.

I'll run a test training with `--sequential-training` and report the result.

**Comment by amitdo** [comment]
Stefan, you can use the parameter max_image_MB to limit the maximum amount of memory used for caching. The default value is 6000.

**Comment by stweil** [comment]
I am not sure whether that parameter is really useful. If I had limited the memory cache to 2 GiB for a data size of 4 GiB, then the cache would be filled and all cache entries would be replaced by new ones before they were used.

**Comment by amitdo** [comment]
It is useful in the case of training with many lines per images.

For one line per image, I agree it's probably not so useful.

**Comment by Shreeshrii** [comment]
https://github.com/tesseract-ocr/tesseract/blob/944c1d9bffe59599924f3c04df4ae0c1e9852373/src/ccstruct/imagedata.h#L40-54

// Enum to determine the caching and data sequencing strategy.
enum CachingStrategy {
  // Reads all of one file before moving on to the next. Requires samples to be
  // shuffled across files. Uses the count of samples in the first file as
  // the count in all the files to achieve high-speed random access. As a
  // consequence, if subsequent files are smaller, they get entries used more
  // than once, and if subsequent files are larger, some entries are not used.
  // Best for larger data sets that don't fit in memory.
  CS_SEQUENTIAL,
  // Reads one sample from each file in rotation. Does not require shuffled
  // samples, but is extremely disk-intensive. Samples in smaller files also
  // get used more often than samples in larger files.
  // Best for smaller data sets that mostly fit in memory.
  CS_ROUND_ROBIN,
};
